src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const splitter = typeof separator === 'string' ? separator : separator + '';
+    let joined = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i] === null || this[i] === undefined ? '' : this[i];
+
+      joined += element;
+
+      if (i !== this.length - 1) {
+        joined += splitter;
+      }
```

Comment:
Try to come up with solution where you don't need to check current index on each iteration